phoenix-core/src/it/java/org/apache/phoenix/end2end/DateTimeIT.java

```diff
@@ -1939,6 +1941,99 @@ public void testDateFormatTimeZone()throws Exception {
+
+        stmt.execute("UPSERT INTO " + tableName + " VALUES(2, TO_DATE('" + date2
+                + "'), TO_TIME('" + date2 + "'), TO_TIMESTAMP('" + date2 + "'))");
+
+        String f = " GMT', 'yyyy-MM-dd HH:mm:ss.SSS z', 'UTC";
+        stmt.execute("UPSERT INTO " + tableName + " VALUES(4, TO_DATE('" + date2 + f
+                + "'), TO_TIME('" + date2 + f + "'), TO_TIMESTAMP('" + date2 + f + "'))");
+
+        PreparedStatement pstmt = conn.prepareStatement("UPSERT INTO " + tableName + " values (?, ?, ?, ?)");
+        pstmt.setInt(1, 3);
+        Timestamp t =  new Timestamp(DateUtil.parseDate(date3).getTime());
+        pstmt.setDate(2, new Date(t.getTime()));
+        pstmt.setTime(3, new Time(t.getTime()));
+        pstmt.setTimestamp(4, t);
+        pstmt.execute();
+        conn.commit();
+
+        ResultSet rs = stmt.executeQuery("SELECT * FROM " + tableName + " ORDER BY id");
+
+        int assertions = 0;
+        for (int i = 0; i < dates.size(); i++){
+            assertTrue(rs.next());
+            String actualDate = dates.get(i);
+            Timestamp parseTimestamp = DateUtil.parseTimestamp(actualDate);
+
+            assertEquals(i + 1, rs.getInt(1));
+            Date expectedDate = DateUtil.getSQLDateInISOChronology(parseTimestamp);
+            assertEquals(expectedDate, rs.getDate(2));
+            assertEquals(rs.getDate(2), rs.getObject(2));
+            assertEquals(actualDate, rs.getString(2));
+
+            Time expectedTime = DateUtil.getSQLTimeInISOChronology(parseTimestamp);
+            assertEquals(new Timestamp(expectedTime.getTime()), new Timestamp(rs.getTime(3).getTime()));
+            assertEquals(expectedTime, rs.getTime(3));
+            assertEquals(rs.getTime(3), rs.getObject(3));
+            assertEquals(actualDate, rs.getString(3));
+
+            Timestamp expectedTimestamp = DateUtil.getSQLTimestampInISOChronology(parseTimestamp);
+            assertEquals(expectedTimestamp, rs.getTimestamp(4));
+            assertEquals(rs.getTimestamp(4), rs.getObject(4));
+            assertEquals(actualDate, rs.getString(4));
+            assertions++;
+        }
+
+        String query = "SELECT year(timestamp), month(timestamp), dayofmonth(timestamp),"
+                + " hour(timestamp), minute(timestamp), second(timestamp) FROM "  + tableName
+                + " ORDER BY id";
+        rs = stmt.executeQuery(query);
+
+        for (int i = 0; i < dates.size(); i++){
+            assertTrue(rs.next());
+            for (int j = 0; j < 6; j++){
+                int expected = Integer.parseInt(dateLists.get(i).get(j));
+                int value = rs.getInt(j + 1);
+                String readFunc = query.split("\\s+")[j + 1];
+                assertTrue("Expected for " + readFunc.substring(0, readFunc.length()-1) + ": " + expected + ", got: " + value,
+                        expected == value);
+                assertions++;
+            }
+        }
+        assertEquals(assertions, 28);
```

Comment:
nit: make this `dates.size() *7`? Not sure if there's a lot of value directly in this assertion either way.

Reply:
It can be deleted, I agree it does not add any value anymore. In development I used it to validate if I do the check on every column of every row.